diff --git public/sites/all/modules/contrib/view_mode_selector/view_mode_selector.module public/sites/all/modules/contrib/view_mode_selector/view_mode_selector.module
index b1718f3..c7915e1 100644
--- public/sites/all/modules/contrib/view_mode_selector/view_mode_selector.module
+++ public/sites/all/modules/contrib/view_mode_selector/view_mode_selector.module
@@ -185,11 +185,8 @@ function view_mode_selector_field_widget_form(&$form, &$form_state, $field, $ins
       unset($element['value']['#options'][$view_mode]);
     }
 
-    // Sort Display Suite view modes by number of regions.
-    function sortByRegionCount($a, $b) {
-      return count($a['layout']['settings']['regions']) - count($b['layout']['settings']['regions']);
-    }
-    uasort($ds_view_modes, 'sortByRegionCount');
+
+    uasort($ds_view_modes, '_view_mode_selector_sort_by_region_count');
 
     // Append the sorted view modes to the field options.
     foreach ($ds_view_modes as $view_mode => $ds_view_mode) {
@@ -406,3 +403,8 @@ function view_mode_selector_view_modes($entity_type) {
 
   return $view_modes;
 }
+
+// Sort Display Suite view modes by number of regions.
+function _view_mode_selector_sort_by_region_count($a, $b) {
+  return count($a['layout']['settings']['regions']) - count($b['layout']['settings']['regions']);
+}
