diff --git a/includes/lr_social_login.functions.inc b/includes/lr_social_login.functions.inc
index 6ec60f4..9dce6f2 100755
--- a/includes/lr_social_login.functions.inc
+++ b/includes/lr_social_login.functions.inc
@@ -191,18 +191,18 @@ function lr_social_login_get_callback_url() {
     }
   }
 
+  // set query to use "destination" path for destination
+  $query = array('destination' => $callback);
+
   // Allow for the login_destination module to insert it's 'current' url query
   if (module_exists('login_destination')) {
     $destination_parse = drupal_parse_url($destination['destination']);
 
     if ($destination_parse['path'] == 'user/login' && !empty($destination_parse['query']) && !empty($destination_parse['query']['current'])) {
-      $query[] = array('destination' => $destination_parse['query']['current']);
-
+      // update query to use "current" path for destination
+      $query = array('destination' => $destination_parse['query']['current']);
     }
   }
-  else {
-    $query[] = array('destination' => $callback);
-  }
 
   $url = url('lr_social_login/token_handler', array(
     'query' => $query,
