diff --git a/public/sites/all/modules/contrib/lr_social_login/includes/lr_social_login.admin.inc b/public/sites/all/modules/contrib/lr_social_login/includes/lr_social_login.admin.inc
index 34998e9..9c39809 100755
--- a/public/sites/all/modules/contrib/lr_social_login/includes/lr_social_login.admin.inc
+++ b/public/sites/all/modules/contrib/lr_social_login/includes/lr_social_login.admin.inc
@@ -107,15 +107,30 @@ function lr_social_login_admin_settings($form, &$form_state) {
       2 => t('Email Address [Ex: johndoe@abc.com ]'),
     )
   );
+
+  $login_redirect_options = array(
+    0 => t('Redirect to same page of site'),
+    1 => t('Redirect to profile page of site'),
+    2 => t('Redirect to custom page of site (If you want user to be redirected to specific URL after login)')
+  );
+
+  $login_register_redirect_options =  array(
+    0 => t('Redirect to same page of site'),
+    1 => t('Redirect to profile page of site'),
+    2 => t('Redirect to custom page of site (If you want user to be redirected to specific URL after registration)'),
+  );
+
+  // Check if r4032login module is enabled
+  if (module_exists('login_destination')) {
+    $login_redirect_options[] = t('Enable integration with r4032login module redirect');
+    $login_register_redirect_options[] = t('Enable integration with r4032login module redirect');
+  }
+
   $form[$basic_group]['lr_social_login_userlogin_redirect'] = array(
     '#type' => 'radios',
     '#title' => t('Redirection settings after login'),
     '#default_value' => variable_get('lr_social_login_userlogin_redirect', 0),
-    '#options' => array(
-      0 => t('Redirect to same page of site'),
-      1 => t('Redirect to profile page of site'),
-      2 => t('Redirect to custom page of site (If you want user to be redirected to specific URL after login)'),
-    ),
+    '#options' => $login_redirect_options
   );
   $form[$basic_group]['lr_social_login_userlogin_redirect']['lr_social_login_custom_redirection'] = array(
     '#type' => 'textfield',
@@ -126,11 +141,7 @@ function lr_social_login_admin_settings($form, &$form_state) {
     '#type' => 'radios',
     '#title' => t('Redirection settings after registration'),
     '#default_value' => variable_get('lr_social_login_userregister_redirect', 0),
-    '#options' => array(
-      0 => t('Redirect to same page of site'),
-      1 => t('Redirect to profile page of site'),
-      2 => t('Redirect to custom page of site (If you want user to be redirected to specific URL after registration)'),
-    ),
+    '#options' => $login_register_redirect_options
   );
   $form[$basic_group]['lr_social_login_userregister_redirect']['lr_social_login_custom_register_redirection'] = array(
     '#type' => 'textfield',
diff --git a/public/sites/all/modules/contrib/lr_social_login/includes/lr_social_login.functions.inc b/public/sites/all/modules/contrib/lr_social_login/includes/lr_social_login.functions.inc
index 8481fd5..9ca8629 100755
--- a/public/sites/all/modules/contrib/lr_social_login/includes/lr_social_login.functions.inc
+++ b/public/sites/all/modules/contrib/lr_social_login/includes/lr_social_login.functions.inc
@@ -503,6 +503,16 @@ function lr_social_login_user_redirect($form, $account, $userprofile, $variable_
       else {
         drupal_redirect_form($form, $redirect = NULL);
       }
+    } elseif (variable_get($variable_path) == 3) {
+      // Redirect to original destination
+      $destination = drupal_get_destination();
+
+      if (!empty($destination['destination'])) {
+        drupal_goto($destination['destination']);
+      }
+      else {
+        drupal_redirect_form($form, $redirect = NULL);
+      }
     }
    else {
           // Redirect to same page.
